hook-common/src/webhook.rs

```diff
@@ -134,3 +134,22 @@ pub struct WebhookJobMetadata {
     pub plugin_id: Option<i32>,
     pub plugin_config_id: Option<i32>,
 }
+
+/// An error originating during a Webhook Job invocation.
+#[derive(Deserialize, Serialize, Debug, PartialEq, Clone)]
+pub struct WebhookJobError {
+    pub r#type: String,
```

Comment:
TIL `r#type`, I had gone with:

```
    #[serde(rename = "type", serialize_with = "serialize_type")]
    pub type_: PluginLogEntryType,
```

Do you have to call it `r#type` when you use it, too?

Reply:
Yeah, it's required everywhere, e.g.:
```
    let r#type = "My type".to_string();
    let error = WebhookJobError {
        r#type,    
    };
    println!("{}", error.r#type);
```

Otherwise the compiler will scream at you that type is a keyword.

I used it as an alternative to `rename`-ing an alias, as:
```
    let json_str = serde_json::to_string(&error).unwrap();
    println!("{}", json_str);
```

Outputs:
```
{"type":"My type"}
```

Reply:
I do not have a strong preference for one or the other by the way, happy if we agree on an alias otherwise; `ty` is a common one for `type`. `r#type` just came out first when writing this.

Reply:
I don't care at all, I just hadn't seen it (or certainly I forgot it).